### ISSUE TITLE
Llama 3.2 Vision 11B on Mac:  Error "thread '<unnamed>' panicked at mistralrs-core/src/sampler.rs" while inferring image

### ISSUE THREAD

**Issue by mathav95raj** [EricLBuehler/mistral.rs#808]
## Describe the bug
On mac M3 max facing the below errors on running llama 3.2 vision 11 B.

**CPU:**
```
mistral.rs % cargo run --release -- -i --isq Q4K vision-plain -m lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k -a vllama 
   Compiling mistralrs-core v0.3.1 (/Users/391080/mistral.rs/mistralrs-core)
   Compiling mistralrs-server v0.3.1 (/Users/391080/mistral.rs/mistralrs-server)
    Finished `release` profile [optimized] target(s) in 26.96s
     Running `target/release/mistralrs-server -i --isq Q4K vision-plain -m lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k -a vllama`
2024-09-30T15:43:41.087938Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-09-30T15:43:41.088041Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-09-30T15:43:41.088152Z  INFO mistralrs_server: Model kind is: normal (no quant, no adapters)
2024-09-30T15:43:41.090090Z  INFO mistralrs_core::pipeline::vision: Loading `tokenizer.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-09-30T15:43:41.090455Z  INFO mistralrs_core::pipeline::vision: Loading `config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-09-30T15:43:41.370605Z  INFO mistralrs_core::pipeline::paths: Found model weight filenames ["model-00001-of-00005.safetensors", "model-00002-of-00005.safetensors", "model-00003-of-00005.safetensors", "model-00004-of-00005.safetensors", "model-00005-of-00005.safetensors"]
2024-09-30T15:43:41.627433Z  INFO mistralrs_core::pipeline::vision: Loading `generation_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-09-30T15:43:41.871300Z  INFO mistralrs_core::pipeline::vision: Loading `preprocessor_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-09-30T15:43:42.332402Z  INFO mistralrs_core::pipeline::vision: Loading `tokenizer_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-09-30T15:43:42.334412Z  INFO mistralrs_core::pipeline::vision: Loading model `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k` on cpu.
2024-09-30T15:43:42.334750Z  INFO mistralrs_core::pipeline::vision: Model config: MLlamaConfig { vision_config: MLlamaVisionConfig { hidden_size: 1280, hidden_act: Gelu, num_hidden_layers: 32, num_global_layers: 8, num_attention_heads: 16, num_channels: 3, intermediate_size: 5120, vision_output_dim: 7680, image_size: 560, patch_size: 14, norm_eps: 1e-5, max_num_tiles: 4, intermediate_layers_indices: [3, 7, 15, 23, 30], supported_aspect_ratios: [(1, 1), (1, 2), (1, 3), (1, 4), (2, 1), (2, 2), (3, 1), (4, 1)] }, text_config: MLlamaTextConfig { rope_scaling: Some(MLlamaRopeScaling { rope_type: Llama3, factor: Some(8.0), original_max_position_embeddings: 8192, attention_factor: None, beta_fast: None, beta_slow: None, short_factor: None, long_factor: None, low_freq_factor: Some(1.0), high_freq_factor: Some(4.0) }), vocab_size: 128256, hidden_size: 4096, hidden_act: Silu, num_hidden_layers: 40, num_attention_heads: 32, num_key_value_heads: 8, intermediate_size: 14336, rope_theta: 500000.0, rms_norm_eps: 1e-5, max_position_embeddings: 131072, tie_word_embeddings: false, cross_attention_layers: [3, 8, 13, 18, 23, 28, 33, 38], use_flash_attn: false, quantization_config: None } }
2024-09-30T15:43:42.336262Z  INFO mistralrs_core::utils::normal: DType selected is F16.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:00<00:00, 13.71it/s]100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:13<00:00, 8.91it/s]100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 108/108 [00:16<00:00, 6.89it/s]100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 111/111 [00:16<00:00, 15.92it/s]100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 560/560 [00:16<00:00, 191.96it/s]
2024-09-30T15:44:04.495389Z  INFO mistralrs_core::pipeline::isq: Applying in-situ quantization into Some(Q4K) to 224 tensors.
2024-09-30T15:44:04.495687Z  INFO mistralrs_core::pipeline::isq: Applying ISQ on 14 threads.
2024-09-30T15:44:11.705717Z  INFO mistralrs_core::pipeline::isq: Applied in-situ quantization into Some(Q4K) to 224 tensors out of 224 total tensors. Took 7.21s
2024-09-30T15:44:11.851567Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<|begin_of_text|>", eos_toks = "<|eot_id|>", "<|end_of_text|>", "<|eom_id|>", unk_tok = `None`
2024-09-30T15:44:11.862064Z  INFO mistralrs_server: Model loaded.
2024-09-30T15:44:11.862156Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
====================
Welcome to interactive mode! Because this model is a vision model, you can enter prompts and chat with the model.

To specify a message with an image, use the `\image` command detailed below.

Commands:
- `\help`: Display this message.
- `\exit`: Quit interactive mode.
- `\system <system message here>`:
    Add a system message to the chat without running the model.
    Ex: `\system Always respond as a pirate.`
- `\image <image URL or local path here> <message here>`: 
    Add a message paired with an image. You are responsible for prefixing the message with anything the model
    requires.
    Ex: `\image path/to/image.jpg Describe what is in this image.`
====================
> hello
Hello! How can I assist you today?
> \image https://upload.wikimedia.org/wikipedia/commons/f/fd/Pink_flower.jpg <|image|>What type of flower is this?
thread '<unnamed>' panicked at mistralrs-core/src/sampler.rs:474:72:
No ordering.
stack backtrace:
   0:        0x10553540c - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h243268f17d714c7f
   1:        0x105556524 - core::fmt::write::hb3cfb8a30e72d7ff
   2:        0x1055312b4 - std::io::Write::write_fmt::hfb2314975de9ecf1
   3:        0x1055364e8 - std::panicking::default_hook::{{closure}}::h14c7718ccf39d316
   4:        0x10553610c - std::panicking::default_hook::hc62e60da3be2f352
   5:        0x105536fac - std::panicking::rust_panic_with_hook::h09e8a656f11e82b2
   6:        0x1055368fc - std::panicking::begin_panic_handler::{{closure}}::h1230eb3cc91b241c
   7:        0x105535898 - std::sys::backtrace::__rust_end_short_backtrace::hc3491307aceda2c2
   8:        0x1055365c4 - _rust_begin_unwind
   9:        0x1055dcfd0 - core::panicking::panic_fmt::ha4b80a05b9fff47a
  10:        0x1055dcfa0 - core::option::expect_failed::h5493e81bb7eff5a4
  11:        0x104b2c064 - core::slice::sort::unstable::ipnsort::hbad9477fa06fb23a
  12:        0x104b412a8 - mistralrs_core::sampler::Sampler::sample_top_kp_min_p::h1228ea6d6021544a
  13:        0x104b420c8 - mistralrs_core::sampler::Sampler::sample::hf8237ceac25cbb3a
  14:        0x1049c2dc8 - mistralrs_core::pipeline::sampling::sample_sequence::{{closure}}::h496c5443bb0ad1c6
  15:        0x104967810 - <futures_util::future::maybe_done::MaybeDone<Fut> as core::future::future::Future>::poll::h8e81cb9896a943ad
  16:        0x104b38fdc - <futures_util::future::join_all::JoinAll<F> as core::future::future::Future>::poll::h40a777465a9d1fe6
  17:        0x104b138a4 - <mistralrs_core::pipeline::vision::VisionPipeline as mistralrs_core::pipeline::Pipeline>::sample_causal_gen::{{closure}}::h1a743770b1c90d88
  18:        0x104b14748 - mistralrs_core::pipeline::Pipeline::step::{{closure}}::h139ce58ad873a519
  19:        0x1046c917c - mistralrs_core::engine::Engine::run::{{closure}}::he107324c17b37e8e
  20:        0x1046bf778 - tokio::runtime::context::runtime::enter_runtime::h64ed88a9d2018b0a
  21:        0x1047bf76c - tokio::runtime::runtime::Runtime::block_on::ha4cd701154d3b938
  22:        0x10469a2b0 - std::sys::backtrace::__rust_begin_short_backtrace::hb27f262618fb7e7c
  23:        0x10463eb0c - core::ops::function::FnOnce::call_once{{vtable.shim}}::h0d2321f91f5f64f0
  24:        0x10553b02c - std::sys::pal::unix::thread::Thread::new::thread_start::h1bd1b9c95010bf71
  25:        0x19eabef94 - __pthread_joiner_wake

> 
```
**Metal**

```
> \image https://upload.wikimedia.org/wikipedia/commons/f/fd/Pink_flower.jpg <|image|>What type of flower is this?
2024-09-30T15:42:43.109177Z ERROR mistralrs_core::engine: prompt step - Model failed with error: WithBacktrace { inner: Msg("Metal contiguous affine I64 not implemented"), backtrace: Backtrace [{ fn: "std::backtrace::Backtrace::create" }, { fn: "candle_core::error::Error::bt" }, { fn: "<candle_core::metal_backend::MetalStorage as candle_core::backend::BackendStorage>::affine" }, { fn: "candle_core::tensor::Tensor::affine" }, { fn: "candle_core::tensor::<impl core::ops::arith::Sub<candle_core::tensor::Tensor> for f64>::sub" }, { fn: "mistralrs_core::vision_models::mllama::vision::_prepare_aspect_ratio_attention_mask" }, { fn: "mistralrs_core::vision_models::mllama::vision::MLlamaVisionModel::forward" }, { fn: "<mistralrs_core::vision_models::mllama::MLlamaModel as mistralrs_core::pipeline::loaders::vision_loaders::VisionModel>::forward" }, { fn: "<mistralrs_core::pipeline::vision::VisionPipeline as mistralrs_core::pipeline::Pipeline>::forward_inputs" }, { fn: "mistralrs_core::pipeline::Pipeline::step::{{closure}}" }, { fn: "mistralrs_core::engine::Engine::run::{{closure}}" }, { fn: "mistralrs_core::MistralRs::new::{{closure}}::{{closure}}" }, { fn: "tokio::runtime::runtime::Runtime::block_on" }, { fn: "std::sys::backtrace::__rust_begin_short_backtrace" }, { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}" }, { fn: "std::sys::pal::unix::thread::Thread::new::thread_start" }, { fn: "__pthread_joiner_wake" }] }
2024-09-30T15:42:43.123644Z ERROR mistralrs_server::interactive_mode: Got a model error: "Metal contiguous affine I64 not implemented\n   0: std::backtrace::Backtrace::create\n   1: candle_core::error::Error::bt\n   2: <candle_core::metal_backend::MetalStorage as candle_core::backend::BackendStorage>::affine\n   3: candle_core::tensor::Tensor::affine\n   4: candle_core::tensor::<impl core::ops::arith::Sub<candle_core::tensor::Tensor> for f64>::sub\n   5: mistralrs_core::vision_models::mllama::vision::_prepare_aspect_ratio_attention_mask\n   6: mistralrs_core::vision_models::mllama::vision::MLlamaVisionModel::forward\n   7: <mistralrs_core::vision_models::mllama::MLlamaModel as mistralrs_core::pipeline::loaders::vision_loaders::VisionModel>::forward\n   8: <mistralrs_core::pipeline::vision::VisionPipeline as mistralrs_core::pipeline::Pipeline>::forward_inputs\n   9: mistralrs_core::pipeline::Pipeline::step::{{closure}}\n  10: mistralrs_core::engine::Engine::run::{{closure}}\n  11: mistralrs_core::MistralRs::new::{{closure}}::{{closure}}\n  12: tokio::runtime::runtime::Runtime::block_on\n  13: std::sys::backtrace::__rust_begin_short_backtrace\n  14: core::ops::function::FnOnce::call_once{{vtable.shim}}\n  15: std::sys::pal::unix::thread::Thread::new::thread_start\n  16: __pthread_joiner_wake\n", response: ChatCompletionResponse { id: "1", choices: [Choice { finish_reason: "error", index: 0, message: ResponseMessage { content: Some(""), role: "assistant", tool_calls: [] }, logprobs: None }], created: 1727710963, model: "lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k", system_fingerprint: "local", object: "chat.completion", usage: Usage { completion_tokens: 0, prompt_tokens: 442, total_tokens: 442, avg_tok_per_sec: 3877.1929, avg_prompt_tok_per_sec: inf, avg_compl_tok_per_sec: NaN, total_time_sec: 0.114, total_prompt_time_sec: 0.0, total_completion_time_sec: 0.0 } }
```

On running without ```--release``` it is taking more than 10 minutes and I am still not seeing any output.

## Latest commit or version
ce02618e0ca82e890321c9a5faa75c44b969f85f

**Comment by mathav95raj** [comment]
With the latest version ced6fc0f8e9a84e1c1ee60990a335279552cb0d2 , it seems the error with ```--features metal``` has gone and the same '<unnamed>' thread error is coming. Just curious, as I am not aware much about rust, has this got something to do with parsing image ?

```
cargo run --features metal --release -- -i --isq Q4K vision-plain -m lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k -a vllama
warning: unused import: `Tensor`
  --> mistralrs-core/src/pipeline/isq.rs:11:27
   |
11 | use candle_core::{Device, Tensor};
   |                           ^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused imports: `IntoParallelRefIterator` and `IntoParallelRefMutIterator`
  --> mistralrs-core/src/pipeline/isq.rs:17:52
   |
17 |     IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator,
   |                                                    ^^^^^^^^^^^^^^^^^^^^^^^
18 |     IntoParallelRefMutIterator, ParallelIterator,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused variable: `write_artifacts`
   --> mistralrs-core/src/pipeline/isq.rs:144:9
    |
144 |         write_artifacts: Option<&PathBuf>,
    |         ^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_write_artifacts`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
   --> mistralrs-core/src/pipeline/isq.rs:147:18
    |
147 |             let (mut tensors, mapper) = match organization {
    |                  ----^^^^^^^
    |                  |
    |                  help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default

warning: `mistralrs-core` (lib) generated 4 warnings (run `cargo fix --lib -p mistralrs-core` to apply 3 suggestions)
    Finished `release` profile [optimized] target(s) in 0.81s
     Running `target/release/mistralrs-server -i --isq Q4K vision-plain -m lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k -a vllama`
2024-10-02T04:02:28.393693Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-10-02T04:02:28.393772Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-10-02T04:02:28.393823Z  INFO mistralrs_server: Model kind is: normal (no quant, no adapters)
2024-10-02T04:02:28.394484Z  INFO mistralrs_core::pipeline::vision: Loading `tokenizer.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-10-02T04:02:28.394747Z  INFO mistralrs_core::pipeline::vision: Loading `config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-10-02T04:02:28.688250Z  INFO mistralrs_core::pipeline::paths: Found model weight filenames ["model-00001-of-00005.safetensors", "model-00002-of-00005.safetensors", "model-00003-of-00005.safetensors", "model-00004-of-00005.safetensors", "model-00005-of-00005.safetensors"]
2024-10-02T04:02:28.960498Z  INFO mistralrs_core::pipeline::vision: Loading `generation_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-10-02T04:02:29.191968Z  INFO mistralrs_core::pipeline::vision: Loading `preprocessor_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-10-02T04:02:29.437473Z  INFO mistralrs_core::pipeline::vision: Loading `tokenizer_config.json` at `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k`
2024-10-02T04:02:29.440660Z  INFO mistralrs_core::pipeline::vision: Loading model `lamm-mit/Cephalo-Llama-3.2-11B-Vision-Instruct-128k` on metal[4294968497].
2024-10-02T04:02:29.441177Z  INFO mistralrs_core::pipeline::vision: Model config: MLlamaConfig { vision_config: MLlamaVisionConfig { hidden_size: 1280, hidden_act: Gelu, num_hidden_layers: 32, num_global_layers: 8, num_attention_heads: 16, num_channels: 3, intermediate_size: 5120, vision_output_dim: 7680, image_size: 560, patch_size: 14, norm_eps: 1e-5, max_num_tiles: 4, intermediate_layers_indices: [3, 7, 15, 23, 30], supported_aspect_ratios: [(1, 1), (1, 2), (1, 3), (1, 4), (2, 1), (2, 2), (3, 1), (4, 1)] }, text_config: MLlamaTextConfig { rope_scaling: Some(MLlamaRopeScaling { rope_type: Llama3, factor: Some(8.0), original_max_position_embeddings: 8192, attention_factor: None, beta_fast: None, beta_slow: None, short_factor: None, long_factor: None, low_freq_factor: Some(1.0), high_freq_factor: Some(4.0) }), vocab_size: 128256, hidden_size: 4096, hidden_act: Silu, num_hidden_layers: 40, num_attention_heads: 32, num_key_value_heads: 8, intermediate_size: 14336, rope_theta: 500000.0, rms_norm_eps: 1e-5, max_position_embeddings: 131072, tie_word_embeddings: false, cross_attention_layers: [3, 8, 13, 18, 23, 28, 33, 38], use_flash_attn: false, quantization_config: None } }
2024-10-02T04:02:29.456662Z  INFO mistralrs_core::utils::normal: DType selected is F16.
100%|████████████████████████████████████████| 560/560 [00:09<00:00, 126.94it/s]
100%|█████████████████████████████████████████| 111/111 [00:09<00:00, 10.21it/s]
100%|█████████████████████████████████████████| 108/108 [00:09<00:00, 19.99it/s]
100%|█████████████████████████████████████████| 118/118 [00:10<00:00, 24.83it/s]
100%|██████████████████████████████████████████████| 9/9 [00:03<00:00, 8.05it/s]
2024-10-02T04:03:18.069949Z  INFO mistralrs_core::pipeline::isq: Applying in-situ quantization into Some(Q4K) to 224 tensors.
2024-10-02T04:04:38.160132Z  INFO mistralrs_core::pipeline::isq: Applied in-situ quantization into Some(Q4K) to 224 tensors out of 224 total tensors. Took 80.09s
2024-10-02T04:04:38.292294Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<|begin_of_text|>", eos_toks = "<|eot_id|>", "<|end_of_text|>", "<|eom_id|>", unk_tok = `None`
2024-10-02T04:04:38.301903Z  INFO mistralrs_server: Model loaded.
2024-10-02T04:04:38.302034Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
====================
Welcome to interactive mode! Because this model is a vision model, you can enter prompts and chat with the model.

To specify a message with an image, use the `\image` command detailed below.

Commands:
- `\help`: Display this message.
- `\exit`: Quit interactive mode.
- `\system <system message here>`:
    Add a system message to the chat without running the model.
    Ex: `\system Always respond as a pirate.`
- `\image <image URL or local path here> <message here>`: 
    Add a message paired with an image. You are responsible for prefixing the message with anything the model
    requires.
    Ex: `\image path/to/image.jpg Describe what is in this image.`
====================
> hello
Hello! How can I assist you today?
>  \image ./images.jpeg <|image|>What is in the picture?
thread '<unnamed>' panicked at mistralrs-core/src/sampler.rs:474:72:
No ordering.
stack backtrace:
   0:        0x101fbfd74 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h243268f17d714c7f
   1:        0x101fe0ea0 - core::fmt::write::hb3cfb8a30e72d7ff
   2:        0x101fbbc1c - std::io::Write::write_fmt::hfb2314975de9ecf1
   3:        0x101fc0e50 - std::panicking::default_hook::{{closure}}::h14c7718ccf39d316
   4:        0x101fc0a74 - std::panicking::default_hook::hc62e60da3be2f352
   5:        0x101fc1914 - std::panicking::rust_panic_with_hook::h09e8a656f11e82b2
   6:        0x101fc1264 - std::panicking::begin_panic_handler::{{closure}}::h1230eb3cc91b241c
   7:        0x101fc0200 - std::sys::backtrace::__rust_end_short_backtrace::hc3491307aceda2c2
   8:        0x101fc0f2c - _rust_begin_unwind
   9:        0x102055708 - core::panicking::panic_fmt::ha4b80a05b9fff47a
  10:        0x1020556d8 - core::option::expect_failed::h5493e81bb7eff5a4
  11:        0x1011732f0 - core::slice::sort::unstable::ipnsort::hace4fcf04fd21441
  12:        0x1010dfdc0 - mistralrs_core::sampler::Sampler::sample_top_kp_min_p::hd82854b36b920e98
  13:        0x1010e0bdc - mistralrs_core::sampler::Sampler::sample::hf89b7407cf654556
  14:        0x1011c6fd8 - mistralrs_core::pipeline::sampling::sample_sequence::{{closure}}::h591b32925eca22bc
  15:        0x10115512c - <futures_util::future::maybe_done::MaybeDone<Fut> as core::future::future::Future>::poll::hf828d9aab19e63b0
  16:        0x101386da8 - <futures_util::future::join_all::JoinAll<F> as core::future::future::Future>::poll::h105756bb00e2c714
  17:        0x1012ea428 - <mistralrs_core::pipeline::vision::VisionPipeline as mistralrs_core::pipeline::Pipeline>::sample_causal_gen::{{closure}}::h57e16f8c6b8f387b
  18:        0x1012ee1d8 - mistralrs_core::pipeline::Pipeline::step::{{closure}}::he46730b378d863bf
  19:        0x101323bb8 - mistralrs_core::engine::Engine::run::{{closure}}::h67e06b90a50afc2f
  20:        0x101313468 - tokio::runtime::park::CachedParkThread::block_on::h4f968c4abd4e3a08
  21:        0x10117b9a4 - tokio::runtime::context::runtime::enter_runtime::h798f2dd99ec46cdc
  22:        0x10108acb4 - std::sys::backtrace::__rust_begin_short_backtrace::h6187247df3188279
  23:        0x10128a9cc - core::ops::function::FnOnce::call_once{{vtable.shim}}::h3f00f0707cffc5bd
  24:        0x101fc5994 - std::sys::pal::unix::thread::Thread::new::thread_start::h1bd1b9c95010bf71
  25:        0x19eabef94 - __pthread_joiner_wake
```

**Comment by EricLBuehler** [comment]
@mathav95raj I think the error is indicative of NaNs, which indicates some numerical instability. I've reproduced the error and will add a fix.

**Comment by EricLBuehler** [comment]
@mathav95raj does the model work for now if you use f32? I'm trying to work out a good fix in #818, but using f32 in the meantime should work.

**Comment by EricLBuehler** [comment]
Hi @mathav95raj! The dtype argument sets the storage and computation.

I just merged #820, which compromises by only loading the vision model in 32 bit if otherwise using f16. This seems to work (and it should be faster), can you please confirm?

**Comment by mathav95raj** [comment]
Hi @EricLBuehler sorry for the delay. Yes now it is running without the dtype f32 argument! Good work! Thanks
<img width="1503" alt="image" src="https://github.com/user-attachments/assets/0425b536-e644-4a47-870d-ee793d8cfb3b">
